src/test/java/org/mockitousage/verification/DelayedExecution.java

```diff
@@ -1,54 +1,78 @@
 package org.mockitousage.verification;
 
-import org.mockitousage.IMethods;
+import static java.lang.System.currentTimeMillis;
+import static java.lang.Thread.MAX_PRIORITY;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.concurrent.locks.LockSupport.parkUntil;
 
-import java.util.ArrayList;
-import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.TimeUnit;
 
 class DelayedExecution {
+    private static final int CORE_POOL_SIZE = 3;
+    /**
+     * Defines the number of milliseconds we expecting a Thread might need to unpark, we use this to avoid "oversleeping" while awaiting the deadline for  
+     */
+    private static final long MAX_EXPECTED_OVERSLEEP_MILLIS = 50;
     private final ScheduledExecutorService executor;
-    private final IMethods mock;
-    private final int delay;
-    private final TimeUnit timeUnit;
-    private final ArrayList<Character> invocations = new ArrayList<Character>();
-
-    DelayedExecution(ScheduledExecutorService executor, IMethods mock, int delay, TimeUnit timeUnit) {
-        this.executor = executor;
-        this.mock = mock;
-        this.delay = delay;
-        this.timeUnit = timeUnit;
+
+    public DelayedExecution() {
+        this.executor = newScheduledThreadPool(CORE_POOL_SIZE, maxPrioThreadFactory());
     }
 
-    void recordAsyncCall(char c) {
-        invocations.add(c);
+    public void callAsync(long delay, TimeUnit timeUnit, Runnable r) {
+        long deadline = timeUnit.toMillis(delay) + currentTimeMillis();
```

Comment:
* `parkUntil(long) `does nothing if the deadline is in the past, so i don't see a problem here.
* The deadline is calculated as soon as possible, when it is calculated later e.g. in `delayedExecution` the deadline may not reflect the delay passed to `callAsync(..)` 
. The time between the call and the calculation is not zero, objects are created and code must be executed in the meantime or the current Thread is halted because of GC's and so on.